bundles/org.openhab.binding.sncf/src/main/java/org/openhab/binding/sncf/internal/handler/StationHandler.java

```diff
@@ -0,0 +1,256 @@
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/null-or-empty-station-id");
+            return;
+        }
+
+        if (thing.getProperties().isEmpty() && !discoverAttributes(stationId)) {
+            return;
+        }
+
+        String timezone = thing.getProperties().get(TIMEZONE);
+        if (timezone == null || timezone.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "@text/null-or-empty-timezone");
+            return;
+        }
+
+        zoneOffset = ZoneId.of(timezone).getRules().getOffset(Instant.now()).getId().replace(":", "");
+        scheduleRefresh(ZonedDateTime.now().plusSeconds(2));
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        super.bridgeStatusChanged(bridgeStatusInfo);
+        if (thing.getStatus() == ThingStatus.ONLINE) {
+            scheduleRefresh(ZonedDateTime.now().plusSeconds(2));
+        }
+    }
+
+    private boolean discoverAttributes(String localStation) {
+        SncfBridgeHandler bridgeHandler = getBridgeHandler();
+        if (bridgeHandler != null) {
+            Map<String, String> properties = new HashMap<>();
+            try {
+                bridgeHandler.stopPointDetail(localStation).ifPresent(stopPoint -> {
+                    String stationLoc = String.format("%s,%s", stopPoint.coord.lat, stopPoint.coord.lon);
+                    properties.put(LOCATION, stationLoc);
+                    properties.put(TIMEZONE, stopPoint.stopArea.timezone);
+                    PointType serverLoc = locationProvider.getLocation();
+                    if (serverLoc != null) {
+                        PointType stationLocation = new PointType(stationLoc);
+                        double distance = serverLoc.distanceFrom(stationLocation).doubleValue();
+                        properties.put(DISTANCE, new QuantityType<>(distance, SIUnits.METRE).toString());
+                    }
+                });
+                ThingBuilder thingBuilder = editThing();
+                thingBuilder.withProperties(properties);
+                updateThing(thingBuilder.build());
+                return true;
+            } catch (SncfException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            }
+        }
+        return false;
+    }
+
+    private void scheduleRefresh(@Nullable ZonedDateTime when) {
+        // Ensure we'll try to refresh in one minute if no valid timestamp is provided
+        long wishedDelay = ZonedDateTime.now().until(when != null ? when : ZonedDateTime.now().plusMinutes(1),
+                ChronoUnit.SECONDS);
+        ScheduledFuture<?> job = refreshJob;
+        if (job != null) {
+            long existingDelay = job.getDelay(TimeUnit.SECONDS);
+            if (existingDelay < wishedDelay) {
```

Comment:
Take care, if run by the current job, existingDelay will be 0. In this case, you want to reschedule.

Reply:
Yes, identified and circumvented.

Reply:
Refresh seems to work well now.